numpy/core/src/multiarray/compiled_base.c

```diff
@@ -112,28 +113,30 @@ arr_bincount(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwds)
     }
     len = PyArray_SIZE(lst);
 
-    if (mlength == NULL) {
-        minlength = 0;
-    }
-    else if (mlength == Py_None) {
+    /*
+     * This if/else if can be removed by changing the argspec to O|On above,
+     * once we retire the deprecation
+     */
+    if (mlength == Py_None) {
         /* NumPy 1.14, 2017-06-01 */
         if (DEPRECATE("0 should be passed as minlength instead of None; "
                       "this will error in future.")) {
             goto fail;
         }
-        minlength = 0;
     }
-    else {
+    else if (mlength != NULL) {
         minlength = PyArray_PyIntAsIntp(mlength);
-        if (minlength < 0) {
-            if (!PyErr_Occurred()) {
-                PyErr_SetString(PyExc_ValueError,
-                                "minlength must be non-negative");
-            }
+        if (error_converting(minlength)) {
             goto fail;
         }
     }
 
+    if (minlength < 0) {
+        PyErr_SetString(PyExc_ValueError,
+                        "minlength must be non-negative");
```

Comment:
Non-negative sounds fine to me. That's mathematicians for you. Positive is incorrect. Maybe `>= 0`?